### ISSUE TITLE
ShardedTensor.size() should be equivalent to torch.Tensor.size()

### ISSUE THREAD

**Issue by pritamdamania87** [pytorch/pytorch#65087]
## 🚀 Feature
[ShardedTensor.size()](https://github.com/pytorch/pytorch/blob/master/torch/distributed/_sharded_tensor/api.py#L575) should be equivalent to [torch.Tensor.size()](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/Functions.h#L193)

## Motivation
torch.Tensor.size() provides the following functionality:

```
import torch

t = torch.rand(3, 4)
print(t.size())
print(t.size(0))
print(t.size(1))
```

ShardedTensor.size() does not accept an optional dim to query the size of a particular dim.

## Pitch

Proposed API:

```
ShardedTensor.size(int dim = None) -> Union[torch.Size(), int]
```
```

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @cbalioglu @gcramer23
